pyrobosim/pyrobosim/utils/general.py

```diff
@@ -35,23 +35,24 @@ def get_data_folder():
 class EntityMetadata:
     """Represents metadata about entities, such as locations or objects."""
 
-    def __init__(self, filename=None):
+    def __init__(self, filename=None, metadata_type=None):
```

Comment:
If you end up keeping this new argument, add it to the docstring.

I say "if you end up keeping this" because I don't quite understand why you need sources to  have separate keys for locations vs. objects, but then you also specify a `metadata_type`. I would think only one of these 2 is needed.

Also, should this metadata type end up being an enum instead of a string?

Reply:
You were right,  I simplified the logic now.